### PR TITLE
Helper function that lists where a given VM is used

### PR DESCRIPTION
**Pull request by marmarta**
A helper function to list vm's usage added to the utils.

### CODE REVIEW

File: qubesadmin/utils.py [hunk 1]
```diff
@@ -115,3 +115,29 @@ def updates_vms_status(qvm_collection):
             # "mixed"
             return None
     return status
+
+# Helper function that returns a list of all the places a given VM is used in.
+# Output is a list of tuples (property_holder, property_name), with None as
+# property_holder for global properties
```
Comment (marmarek) [inline qubesadmin/utils.py]: This should be in docstring. pylint told you this already ;)

File: qubesadmin/utils.py [hunk 2]
```diff
@@ -115,3 +115,29 @@ def updates_vms_status(qvm_collection):
...
+
+
+def vm_usage(app, reference_vm):
```
Comment (marmarek) [inline qubesadmin/utils.py]: Could you make this function name more specific? IMO `vm_usage` could be confused with disk usage.

File: qubesadmin/utils.py [hunk 3]
```diff
@@ -115,3 +115,29 @@ def updates_vms_status(qvm_collection):
...
+    for prop in global_properties:
+        if reference_vm == getattr(app, prop, None):
+            result.append([None, prop])
+
+    vm_properties = ['template', 'netvm', 'default_dispvm']
+
+    for vm in app.domains:
+        for prop in vm_properties:
+            if reference_vm == getattr(vm, prop, None) and \
+                    not vm.property_is_default(prop):
```
Comment (marmarek) [inline qubesadmin/utils.py:140]: A little optimization: switch conditions order (most properties will have default value).

**Review by marmarek — CHANGES_REQUESTED** [review]
Besides the comments above, this could really use some unit test (handling various corner cases, like missing property).

**Review by marmarek — APPROVED** [review]
